assets/src/edit-story/components/canvas/canvasElementDropzone.js

```diff
@@ -49,7 +49,18 @@ function CanvasElementDropzone({ children }) {
 
   const onDropHandler = useCallback(
     (e) => {
-      if (isDragType(e, 'resource/media') && !activeDropTargetId) {
+      // Handles onDrop for shapes.
+      if (isDragType(e, 'shape') && !activeDropTargetId) {
+        const props = JSON.parse(e.dataTransfer.getData('shape'));
+        const { x, y } = pageContainer?.getBoundingClientRect();
```

Comment:
Why guard for `null` in `pageContainer?.`? If `null` the whole expression returns `null` and then the destruct fails anyway. If `null` is an option, you need some default alternative:

```js
        const { x, y } = pageContainer?.getBoundingClientRect() || { x: 0, y: 0 };
```

But what happens then? And can it even happen?

Reply:
Done. Removed the null check.